### PR TITLE
Bug 2048450: Bump containerd to 1.4.11

### PR DESCRIPTION
**Pull request by perdasilva**
### Description

Twistlock scan showed that openshift4/ose-operator-registry:v4.9 is vulnerable to [CVE-2020-15257](https://access.redhat.com/security/cve/CVE-2020-15257). As discussed at nvd site https://nvd.nist.gov/vuln/detail/CVE-2020-15257, this vulnerability can be fixed by upgrading the https://github.com/containerd/containerd to v1.4.3 and above.
With this bug ticket, I would want to let you know about this and any possible solution from you.

Version-Release number of selected component (if applicable):v4.9

How reproducible:
Running the twistlock scan on this image will show the vulnerability.

### Solution

Backporting commits that bumped the containerd dependency for 4.10+

### CODE REVIEW

**Review by njhale — CHANGES_REQUESTED** [review]
Something looks a bit off here.

**Review by anik120 — COMMENTED** [review]
/lgtm

File: staging/operator-registry/go.mod [hunk 1]
```diff
@@ -7,7 +7,7 @@ require (
 	github.com/blang/semver v3.5.1+incompatible
 	github.com/bugsnag/bugsnag-go v1.5.3 // indirect
 	github.com/bugsnag/panicwrap v1.2.0 // indirect
-	github.com/containerd/containerd v1.3.2
+	github.com/containerd/containerd v1.4.11
```
[the remark this answers is not in the document]
Comment (njhale) [reply staging/operator-registry/go.mod:10]: /hold

It looks like it's still being replaced at the root with @ecordell's fork.

File: staging/operator-registry/go.mod [hunk 2]
```diff
@@ -66,13 +65,5 @@ require (
...
-	// 1. stage registry proxy didn't like requests with no scopes, see https://github.com/containerd/containerd/pull/4223
-	// 2. prod registry proxy returns a 403 on post, see https://github.com/containerd/containerd/pull/3913
-	// The fork can be removed when both issues are resolved in a release, which should be 1.4.0
-	github.com/containerd/containerd => github.com/ecordell/containerd v1.3.1-0.20200629153125-0ff1a1be2fa5
-
-	// latest tag resolves to a very old version. this is only used for spinning up local test registries
-	github.com/docker/distribution => github.com/docker/distribution v0.0.0-20191216044856-a8371794149d
-)
+// latest tag resolves to a very old version. this is only used for spinning up local test registries
+replace github.com/docker/distribution => github.com/docker/distribution v0.0.0-20191216044856-a8371794149d
```
Comment (perdasilva) [inline staging/operator-registry/go.mod:69]: @njhale should we drop this for this backport and revert this to the original?

Comment (njhale) [reply staging/operator-registry/go.mod:69]: @perdasilva the cherry-pick is fine -- we should update the root go.mod and call `make vendor` from the root directory.

File: staging/operator-registry/go.mod (hunk 1 above)
[the remark this answers is not in the document]
Comment (njhale) [reply staging/operator-registry/go.mod:10]: > Does CI recreate the vendor directory?

No, you need to call `make vendor` when you cherry-pick from the root. `replaces` has special semantics as well; only the `replaces` at the root level is observed (that rule is for go modules in general too).

[the remark this answers is not in the document]
Comment (timflannagan) [reply staging/operator-registry/go.mod:10]: There should be automation that catches an out-of-date root vendor directory when running `make verify` but it's possible we didn't add that verification target in the release-4.9 branch?

**Review by dinhxuanvu — COMMENTED** [review]
/lgtm

**Pull request merged**